floater/utils.py

```diff
@@ -218,3 +218,61 @@ def floats_to_castra(input_dir, output_fname, progress=False, **kwargs):
         if not c:
             c = Castra(output_fname, template=df)
         c.extend(df)
+
+def floats_to_netcdf(input_dir, output_fname,
+                     float_file_prefix,
+                     ref_time, step_time=86400):
+    """Convert MITgcm float data to NetCDF format.
+
+    Parameters
+    ----------
+    input_dir : path
+        Where to find the MITgcm output data
+    output_fname : path
+        Filename of the NetCDF data store
+    float_file_prefix: str
+        Prefix of MITgcm output files
+    ref_time: str
+        Reference time, format: YYYY-MM-DD
+    step_time: int
+        Step time, unit: second
+    """
+    import dask.dataframe as dd
+    import xarray as xr
+    from glob import glob
+
+    output_fname = _maybe_add_suffix(output_fname, '.nc')
+
+    float_files = glob(float_file_prefix+'.*.csv')
+    float_digits = 10
+    step_code = len(float_file_prefix) + float_digits + 1
+    float_timesteps = sorted(list({float_file[:step_code] for float_file in float_files}))
+
+    float_columns = ['npart', 'time', 'x', 'y', 'z', 'u', 'v', 'vort']
+    var_names = float_columns[2:]
+
+    var_int = [('npart', np.int32)]
+    var_float = [(var, np.float32) for var in float_columns[1:]]
+    float_dtypes = np.dtype(var_int+var_float)
+
+    for float_timestep in float_timesteps:
+        input_path = input_dir + float_timestep + '.*.csv'
+        df = dd.read_csv(input_path, names=float_columns, dtype=float_dtypes, header=None)
+        dfc = df.compute()
+        dfcs = dfc.sort('npart')
+        step_num = int(dfcs.time.values[0])//step_time
+        if ref_time is not None:
+            ref_time = np.datetime64(ref_time, 'ns')
+            del_time = np.timedelta64(step_time*step_num, 's')
+            time = np.array([ref_time+del_time])
+        else:
+            time = np.array([np.int32(step_num)])
+        npart = dfcs.npart.values
+        var_shape = (1, len(npart))
+        data_vars = {var_name: (['time', 'npart'], dfcs[var_name].values.reshape(var_shape)) for var_name in var_names}
+        ds = xr.Dataset(data_vars, coords={'time': time, 'npart': npart})
+        output_dir = input_dir + output_fname + '/'
```

Comment:
I still finding this a bit confusing since `output_fname` is actually a directory. I would also prefer if we did not force the user to nest `output_dir` in `input_dir` (especially because `input_dir` already has 100's of files). What do you folks think?